src/routes/key_custodian.rs

```diff
@@ -114,12 +114,15 @@ pub async fn decrypt(
             );
             aes_decrypt_custodian_key(&mut tenant_config, inner_key1, inner_key2).await?;
 
-            let tenant_app_state =
-                TenantAppState::new(global_app_state.global_config.clone(), tenant_config)
-                    .await
-                    .change_context(error::ApiError::TenantError(
-                        "Failed while creating AppState for tenant",
-                    ))?;
+            let tenant_app_state = TenantAppState::new(
+                global_app_state.global_config.clone(),
```

Comment:
The global config is starting to become bulky, we should wrap it with a shared reference.